code/game/gamemodes/objective_holder.dm

```diff
@@ -59,9 +59,10 @@
  * Replace old_objective with new_objective
  */
 /datum/objective_holder/proc/replace_objective(datum/objective/old_objective, datum/objective/new_objective)
-	new_objective = add_objective(new_objective, add_to_list = FALSE)
-	new_objective.owner = old_objective.owner
+	new_objective = add_objective(new_objective, add_to_list = FALSE, owner_override = old_objective.owner)
 	new_objective.team = old_objective.team
```

Comment:
you'd need to do this for the team too. But now that I think about it, it may be easier to make sure `new_objective` is an initalized type and pre-assign the owner and team before calling `add_objective`

Reply:
Done! PTAL